core/src/traits/Kokkos_StaticBatchSizeTrait.hpp

```diff
@@ -0,0 +1,87 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_STATIC_BATCH_SIZE_TRAIT_HPP
+#define KOKKOS_STATIC_BATCH_SIZE_TRAIT_HPP
+
+#include <Kokkos_Macros.hpp>
+#include <traits/Kokkos_PolicyTraitAdaptor.hpp>
+#include <traits/Kokkos_Traits_fwd.hpp>
+
+namespace Kokkos::Experimental {
+
+template <unsigned int BatchSize = 1>
+struct StaticBatchSize {
+  using static_batch_size = StaticBatchSize;
+  using type              = StaticBatchSize<BatchSize>;
+  constexpr static unsigned int batch_size{
+      BatchSize};  // Default batch size is 1
```

Comment:
Not blocking but it's redundant with the default template argument value.
```suggestion
      BatchSize};
```